deploy/eck-elasticsearch/templates/elasticsearch.yaml

```diff
@@ -0,0 +1,41 @@
+---
+apiVersion: elasticsearch.k8s.elastic.co/v1
+kind: Elasticsearch
+metadata:
+  name: {{ required "A name for the Elasticsearch instance is required" .Values.name }}
+  namespace: {{ .Release.Namespace }}
+  {{- if .Values.labels }}
+  labels:
+    {{- toYaml .Values.labels | indent 4 }}
+  {{- end }}
+  {{- if .Values.annotations }}
+  annotations:
+    {{- toYaml .Values.annotations | indent 4 }}
```

Comment:
I think we should also include the license annotation proposed here https://github.com/elastic/cloud-on-k8s/pull/5815